include/onnxruntime/core/session/experimental_onnxruntime_cxx_api.h

```diff
@@ -47,6 +47,19 @@ struct Session : Ort::Session {
   std::vector<std::vector<int64_t> > GetOverridableInitializerShapes() const;
 };
 
+struct Value : Ort::Value {
+  Value(OrtValue* p)
+      : Ort::Value(p){};
+
+  template <typename T>
+  static Ort::Value CreateTensor(T* p_data, size_t p_data_element_count, const std::vector<int64_t>& shape);
```

Comment:
Why not simplify it even further by accepting a vector? Then you don't have to pass the element count as well.

Reply:
I'm not sure this can be simplified further. In my particular use case, I'm creating tensors from data that belongs to other libraries (OpenCV and Eigen). It makes sense to simplify the shape arguments here but passing in arguments for a data pointer seems to be the only library agnostic way without making some assumption about how other libraries store their data.